packages/compiler/src/core/name-resolver.ts

```diff
@@ -153,6 +156,8 @@ export function createResolver(program: Program): NameResolver {
   const metaTypePrototypes = createMetaTypePrototypes();
 
   const nullSym = createSymbol(undefined, "null", SymbolFlags.None);
+  const augmentDecoratorsForSym = new Map<Sym, AugmentDecoratorStatementNode[]>();
```

Comment:
When do we use `Sym` vs the symbol id as the key

Reply:
I don't know what symbol id is used for. I know in TypeScript they use symbol id a lot because it predates Maps being in the language...

Reply:
I guess that maybe make sense, we pretty much use it just to keep track of the symbol links otherwise all other maps use the symbol itself as keys. Might be somiething we can cleanup in the future